### PR TITLE
fixed RHO (parent) dot notation parsing in vertical notation

### PR DESCRIPTION
**Pull request by Graur**
Closes: #1341

What's done:
- added new rule to parser;
- fixed dot notation usage with parent (`^`) object in vertical notation;
- added several parsing and runtime tests.

### CODE REVIEW

**Comment by Graur** [comment]
@levBagryansky please, check this one

File: eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml [hunk 1]
```diff
@@ -64,6 +64,9 @@ eo: |
           d.
         e.
           f.
+      ^.
+        ^.
+          c
```
Comment (yegor256) [inline eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @Graur try this:

```
$.
  a.
    b
```

Comment (Graur) [reply eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @yegor256 This fails too

Comment (yegor256) [reply eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @Graur this is exactly my point: your fix is very "narrow". We can do a more generic fix, which will cover all cases (including mine just suggested above).

Comment (Graur) [reply eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @yegor256 I agree, but can we use `DOT` for all `head`? I corrected code for your comment as I understood it, check please.

Comment (yegor256) [reply eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @Graur you just did it)

Comment (Graur) [reply eo-parser/src/test/resources/org/eolang/parser/packs/full-syntax.yaml:69]: @yegor256 Good, let me just add some tests

File: eo-parser/src/main/antlr4/org/eolang/parser/Program.g4 [hunk 2]
```diff
@@ -171,10 +171,13 @@ head
...
     AT
+    DOT?
     |
     RHO
+    DOT?
     |
     XI
+    DOT?
     |
     SIGMA
```
Comment (yegor256) [inline eo-parser/src/main/antlr4/org/eolang/parser/Program.g4:182]: @Graur why not here too? I think that it's better to group them:

```
(
AT
|
RHO
|
XI
|
SIGMA
)
DOT?
```

Comment (Graur) [reply eo-parser/src/main/antlr4/org/eolang/parser/Program.g4:182]: @yegor256 Good idea, thanks!

**Comment by Graur** [comment]
@yegor256 please, take a look one more time

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**